homeassistant/components/hive/light.py

```diff
@@ -137,32 +69,25 @@ async def async_turn_off(self, **kwargs):
         """Instruct the light to turn off."""
         await self.hive.light.turnOff(self.device)
 
-    @property
-    def color_mode(self) -> str:
-        """Return the color mode of the light."""
-        if self.device["hiveType"] == "warmwhitelight":
-            return ColorMode.BRIGHTNESS
-        if self.device["hiveType"] == "tuneablelight":
-            return ColorMode.COLOR_TEMP
-        if self.device["hiveType"] == "colourtuneablelight":
-            if self.device["status"]["mode"] == "COLOUR":
-                return ColorMode.HS
-            return ColorMode.COLOR_TEMP
-        return ColorMode.ONOFF
-
-    @property
-    def supported_color_modes(self) -> set[str] | None:
-        """Flag supported color modes."""
-        if self.device["hiveType"] == "warmwhitelight":
-            return {ColorMode.BRIGHTNESS}
-        if self.device["hiveType"] == "tuneablelight":
-            return {ColorMode.COLOR_TEMP}
-        if self.device["hiveType"] == "colourtuneablelight":
-            return {ColorMode.COLOR_TEMP, ColorMode.HS}
-        return {ColorMode.ONOFF}
-
     async def async_update(self):
         """Update all Node data from Hive."""
         await self.hive.session.updateData(self.device)
         self.device = await self.hive.light.getLight(self.device)
         self.attributes.update(self.device.get("attributes", {}))
+        self._attr_extra_state_attributes = {
+            ATTR_MODE: self.attributes.get(ATTR_MODE),
+        }
+        self._attr_available = self.device["deviceData"].get("online")
+        if self._attr_available:
+            self._attr_is_on = self.device["status"]["state"]
+            self._attr_brightness = self.device["status"]["brightness"]
+            self._attr_min_mireds = self.device.get("min_mireds")
+            self._attr_max_mireds = self.device.get("max_mireds")
+            if self.device["hiveType"] == "warmwhitelight":
+                self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
```

Comment:
The device type is static for the entity lifetime, correct? Then we should set the supported color modes attribute in the entity init method. We don't need to update it on every update.

Reply:
Could we use entity descriptions to define each bulb?

Reply:
Unfortunately the light integration doesn't have its own entity descriptions yet, so that would be a bit cumbersome. Not impossible, but it would require some extra lines to set the attributes in the init method, besides defining a custom entity description class.